### ISSUE TITLE
I would like to compile rsync 3.2.4-dev but I've got 3.2.3-70. Why ?

### ISSUE THREAD

**Issue by Marietto2008** [WayneD/rsync#199]
Hello.

I'm on ubuntu 21.04. I have compiled rsync from source to have the 3.24.dev,but I haven't got it. the version I've got is still 3.2.3.70. I've saved all the commands that I have issued :

```
python3 -mpip install --user commonmark 

nano /root/.bashrc and added "export PATH=$PATH:/root/.local/bin" 

git clone https://github.com/WayneD/rsync.git 

cd rsync

./md2man --test rsync-ssl.1.md = OK

./prepare-source fetchgen 

apt install -y gcc g++ gawk autoconf automake python3-cmarkgfm acl libacl1-dev attr libattr1-dev libxxhash-dev libzstd-dev liblz4-dev libssl-dev

./configure
make
make install
./rsync --version 

rsync  version v3.2.3-70-g9dad3721  protocol version 31
```

IT IS NOT 3.2.4-dev. why ?

**Comment by Marietto2008** [comment]
tried again :

root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync# ./configure

checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... no
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking whether time.h and sys/time.h may both be included... yes
checking for sys/wait.h that is POSIX.1 compatible... yes
checking sys/fcntl.h usability... yes
checking sys/fcntl.h presence... yes
checking for sys/fcntl.h... yes
checking sys/select.h usability... yes
checking sys/select.h presence... yes
checking for sys/select.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/unistd.h usability... yes
checking sys/unistd.h presence... yes
checking for sys/unistd.h... yes
checking for unistd.h... (cached) yes
checking utime.h usability... yes
checking utime.h presence... yes
checking for utime.h... yes
checking compat.h usability... no
checking compat.h presence... no
checking for compat.h... no
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking ctype.h usability... yes
checking ctype.h presence... yes
checking for ctype.h... yes
checking for sys/wait.h... (cached) yes
checking for sys/stat.h... (cached) yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/filio.h usability... no
checking sys/filio.h presence... no
checking for sys/filio.h... no
checking for string.h... (cached) yes
checking for stdlib.h... (cached) yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking sys/mode.h usability... no
checking sys/mode.h presence... no
checking for sys/mode.h... no
checking grp.h usability... yes
checking grp.h presence... yes
checking for grp.h... yes
checking sys/un.h usability... yes
checking sys/un.h presence... yes
checking for sys/un.h... yes
checking sys/attr.h usability... no
checking sys/attr.h presence... no
checking for sys/attr.h... no
checking arpa/inet.h usability... yes
checking arpa/inet.h presence... yes
checking for arpa/inet.h... yes
checking arpa/nameser.h usability... yes
checking arpa/nameser.h presence... yes
checking for arpa/nameser.h... yes
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking for sys/types.h... (cached) yes
checking netdb.h usability... yes
checking netdb.h presence... yes
checking for netdb.h... yes
checking malloc.h usability... yes
checking malloc.h presence... yes
checking for malloc.h... yes
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking iconv.h usability... yes
checking iconv.h presence... yes
checking for iconv.h... yes
checking libcharset.h usability... no
checking libcharset.h presence... no
checking for libcharset.h... no
checking langinfo.h usability... yes
checking langinfo.h presence... yes
checking for langinfo.h... yes
checking mcheck.h usability... yes
checking mcheck.h presence... yes
checking for mcheck.h... yes
checking sys/acl.h usability... yes
checking sys/acl.h presence... yes
checking for sys/acl.h... yes
checking acl/libacl.h usability... yes
checking acl/libacl.h presence... yes
checking for acl/libacl.h... yes
checking attr/xattr.h usability... yes
checking attr/xattr.h presence... yes
checking for attr/xattr.h... yes
checking sys/xattr.h usability... yes
checking sys/xattr.h presence... yes
checking for sys/xattr.h... yes
checking sys/extattr.h usability... no
checking sys/extattr.h presence... no
checking for sys/extattr.h... no
checking dl.h usability... no
checking dl.h presence... no
checking for dl.h... no
checking popt.h usability... no
checking popt.h presence... no
checking for popt.h... no
checking popt/popt.h usability... yes
checking popt/popt.h presence... yes
checking for popt/popt.h... yes
checking linux/falloc.h usability... yes
checking linux/falloc.h presence... yes
checking for linux/falloc.h... yes
checking netinet/in_systm.h usability... yes
checking netinet/in_systm.h presence... yes
checking for netinet/in_systm.h... yes
checking netgroup.h usability... no
checking netgroup.h presence... no
checking for netgroup.h... no
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking xxhash.h usability... yes
checking xxhash.h presence... yes
checking for xxhash.h... yes
checking openssl/md4.h usability... yes
checking openssl/md4.h presence... yes
checking for openssl/md4.h... yes
checking openssl/md5.h usability... yes
checking openssl/md5.h presence... yes
checking for openssl/md5.h... yes
checking zstd.h usability... yes
checking zstd.h presence... yes
checking for zstd.h... yes
checking lz4.h usability... yes
checking lz4.h presence... yes
checking for lz4.h... yes
checking sys/file.h usability... yes
checking sys/file.h presence... yes
checking for sys/file.h... yes
checking for netinet/ip.h... yes
checking whether sys/types.h defines makedev... no
checking sys/mkdev.h usability... no
checking sys/mkdev.h presence... no
checking for sys/mkdev.h... no
checking sys/sysmacros.h usability... yes
checking sys/sysmacros.h presence... yes
checking for sys/sysmacros.h... yes
configure.sh: Configuring rsync 3.2.4dev
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking whether to include debugging symbols... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking how to run the C preprocessor... gcc -E
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking for gawk... gawk
checking for egrep... (cached) /usr/bin/grep -E
checking for a BSD-compatible install... /usr/bin/install -c
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gcc option to accept ISO C99... none needed
checking for gcc option to accept ISO Standard C... (cached) none needed
checking for perl... /usr/bin/perl
checking for python3... /usr/bin/python3
checking if md2man can create man pages... yes
checking if we require man-page building... optional
checking for remsh... 0
checking for sh... /usr/bin/sh
checking for fakeroot... /usr/bin/fakeroot
checking the group for user "nobody"... nogroup
checking whether to enable SIMD optimizations... yes (x86_64)
checking if assembler accepts noexecstack... yes
checking whether to enable ASM optimizations... yes (x86_64)
checking for broken largefile support... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking ipv6 stack type... linux-glibc
checking for library containing getaddrinfo... none required
checking whether to call shutdown on all sockets... no
checking whether to enable use of openssl crypto library... yes
checking for library containing MD5_Init... -lcrypto
checking whether to enable xxhash checksum support... yes
checking for library containing XXH64_createState... -lxxhash
checking whether to enable zstd compression... yes
checking for library containing ZSTD_minCLevel... -lzstd
checking whether to enable LZ4 compression... yes
checking for library containing LZ4_compress_default... -llz4
checking if makedev takes 3 args... no
checking size of int... 4
checking size of long... 8
checking size of long long... 8
checking size of short... 2
checking size of int16_t... 2
checking size of uint16_t... 2
checking size of int32_t... 4
checking size of uint32_t... 4
checking size of int64_t... 8
checking size of off_t... 8
checking size of off64_t... 8
checking size of time_t... 8
checking size of char*... 8
checking for inline... inline
checking for long double with more range or precision than double... yes
checking for uid_t in sys/types.h... yes
checking for mode_t... yes
checking for off_t... yes
checking for size_t... yes
checking for pid_t... yes
checking for id_t... yes
checking type of array argument to getgroups... gid_t
checking for struct stat.st_rdev... yes
checking for struct stat.st_mtimensec... no
checking for struct stat.st_mtimespec.tv_nsec... no
checking for struct stat.st_mtim.tv_nsec... yes
checking for socklen_t... yes
checking for errno in errno.h... yes
checking for connect... yes
checking for library containing inet_ntop... none required
checking for library containing iconv_open... none required
checking for library containing libiconv_open... no
checking for iconv declaration... 
         extern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);
checking for inet_ntop... yes
checking for inet_pton... yes
checking for struct addrinfo... yes
checking for struct sockaddr_storage... yes
checking whether defines needed by getaddrinfo exist... yes
checking for getaddrinfo... yes
checking for struct sockaddr.sa_len... no
checking for struct sockaddr_in.sin_len... no
checking for struct sockaddr_un.sun_len... no
checking for struct sockaddr_in6.sin6_scope_id... yes
checking for struct stat64... yes
checking for strcasecmp... yes
checking for aclsort... no
checking for aclsort in -lsec... no
checking for utime.h... (cached) yes
checking whether utime accepts a null argument... yes
checking for size_t... (cached) yes
checking for working alloca.h... yes
checking for alloca... yes
checking for waitpid... yes
checking for wait4... yes
checking for getcwd... yes
checking for chown... yes
checking for chmod... yes
checking for lchmod... yes
checking for mknod... yes
checking for mkfifo... yes
checking for fchmod... yes
checking for fstat... yes
checking for ftruncate... yes
checking for strchr... yes
checking for readlink... yes
checking for link... yes
checking for utime... yes
checking for utimes... yes
checking for lutimes... yes
checking for strftime... yes
checking for chflags... no
checking for getattrlist... no
checking for mktime... yes
checking for innetgr... yes
checking for linkat... yes
checking for memmove... yes
checking for lchown... yes
checking for vsnprintf... yes
checking for snprintf... yes
checking for vasprintf... yes
checking for asprintf... yes
checking for setsid... yes
checking for strpbrk... yes
checking for strlcat... no
checking for strlcpy... no
checking for strtol... yes
checking for mallinfo... yes
checking for getgroups... yes
checking for setgroups... yes
checking for geteuid... yes
checking for getegid... yes
checking for setlocale... yes
checking for setmode... no
checking for open64... yes
checking for lseek64... yes
checking for mkstemp64... yes
checking for mtrace... yes
checking for va_copy... no
checking for __va_copy... no
checking for seteuid... yes
checking for strerror... yes
checking for putenv... yes
checking for iconv_open... yes
checking for locale_charset... no
checking for nl_langinfo... yes
checking for getxattr... yes
checking for extattr_get_link... no
checking for sigaction... yes
checking for sigprocmask... yes
checking for setattrlist... no
checking for getgrouplist... yes
checking for initgroups... yes
checking for utimensat... yes
checking for posix_fallocate... yes
checking for attropen... no
checking for setvbuf... yes
checking for nanosleep... yes
checking for usleep... yes
checking for setenv... yes
checking for unsetenv... yes
checking for useable fallocate... yes
checking for FALLOC_FL_PUNCH_HOLE... yes
checking for FALLOC_FL_ZERO_RANGE... yes
checking for SYS_fallocate... yes
checking whether posix_fallocate is efficient... no
checking for getpgrp... yes
checking for tcgetpgrp... yes
checking whether getpgrp requires zero arguments... yes
checking whether chown() modifies symlinks... no
checking whether link() can hard-link symlinks... yes
checking whether link() can hard-link special files... yes
checking for working socketpair... yes
checking for getpass... yes
checking for poptGetContext in -lpopt... no
checking whether to use included libpopt... ./popt
checking whether to use included zlib... ./zlib
checking for unsigned char... yes
checking for broken readdir... no
checking for utimbuf... yes
checking if gettimeofday takes tz argument... yes
checking for C99 vsnprintf... yes
checking for secure mkstemp... yes
checking if mknod creates FIFOs... yes
checking if mknod creates sockets... yes
checking whether -c -o works... yes
checking for _acl... no
checking for __acl... no
checking for _facl... no
checking for __facl... no
checking whether to support ACLs... running tests:
checking for acl_get_file in -lacl... yes
checking for ACL support... yes
checking ACL test results... Using posix ACLs
checking for acl_get_perm_np... no
checking whether to support extended attributes... Using Linux xattrs
checking for getxattr in -lattr... yes
configure.sh: creating ./config.status
config.status: creating Makefile
config.status: creating lib/dummy
config.status: creating zlib/dummy
config.status: creating popt/dummy
config.status: creating shconfig
config.status: creating config.h

    rsync 3.2.4dev configuration successful

root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync# make

gawk -f ./daemon-parm.awk ./daemon-parm.txt
gawk -f ./mkproto.awk ./*.c ./lib/compat.c daemon-parm.h
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c flist.c -o flist.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c rsync.c -o rsync.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c generator.c -o generator.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c receiver.c -o receiver.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c cleanup.c -o cleanup.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c sender.c -o sender.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c exclude.c -o exclude.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c util1.c -o util1.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c util2.c -o util2.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c main.c -o main.o
main.c: In function ‘show_malloc_stats’:
main.c:473:2: warning: ‘mallinfo’ is deprecated [-Wdeprecated-declarations]
  473 |  mi = mallinfo();
      |  ^~
In file included from rsync.h:359,
                 from main.c:23:
/usr/include/malloc.h:118:24: note: declared here
  118 | extern struct mallinfo mallinfo (void) __THROW __MALLOC_DEPRECATED;
      |                        ^~~~~~~~
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c checksum.c -o checksum.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c match.c -o match.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c syscall.c -o syscall.o
syscall.c: In function ‘do_chmod’:
syscall.c:259:5: warning: ‘code’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  259 |  if (code != 0 && (preserve_perms || preserve_executability))
      |     ^
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c log.c -o log.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c backup.c -o backup.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c delete.c -o delete.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c options.c -o options.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c io.c -o io.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c compat.c -o compat.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c hlink.c -o hlink.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c token.c -o token.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c uidlist.c -o uidlist.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c socket.c -o socket.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c hashtable.c -o hashtable.o
gawk -f ./help-from-md.awk -v hfile=help-rsync.h ./rsync.1.md
gawk -f ./help-from-md.awk -v hfile=help-rsyncd.h ./rsync.1.md
gawk -f ./define-from-md.awk -v hfile=default-cvsignore.h ./rsync.1.md
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c usage.c -o usage.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c fileio.c -o fileio.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c batch.c -o batch.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c clientname.c -o clientname.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c chmod.c -o chmod.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c acls.c -o acls.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c xattrs.c -o xattrs.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c progress.c -o progress.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c pipe.c -o pipe.o
gcc -I. -Wa,--noexecstack -c -o lib/md5-asm-x86_64.o ./lib/md5-asm-x86_64.S
g++ -I. -O2 -fno-exceptions -fno-rtti -c -o simd-checksum-x86_64.o ./simd-checksum-x86_64.cpp
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c params.c -o params.o
gawk -f ./define-from-md.awk -v hfile=default-dont-compress.h ./rsync.1.md
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c loadparm.c -o loadparm.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c clientserver.c -o clientserver.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c access.c -o access.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c connection.c -o connection.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c authenticate.c -o authenticate.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/wildmatch.c -o lib/wildmatch.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/compat.c -o lib/compat.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/snprintf.c -o lib/snprintf.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/mdfour.c -o lib/mdfour.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/md5.c -o lib/md5.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/permstring.c -o lib/permstring.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/pool_alloc.c -o lib/pool_alloc.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/sysacls.c -o lib/sysacls.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c lib/sysxattrs.c -o lib/sysxattrs.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/deflate.c -o zlib/deflate.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/inffast.c -o zlib/inffast.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/inflate.c -o zlib/inflate.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/inftrees.c -o zlib/inftrees.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/trees.c -o zlib/trees.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/zutil.c -o zlib/zutil.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/adler32.c -o zlib/adler32.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/compress.c -o zlib/compress.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c zlib/crc32.c -o zlib/crc32.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c popt/findme.c -o popt/findme.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c popt/popt.c -o popt/popt.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c popt/poptconfig.c -o popt/poptconfig.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c popt/popthelp.c -o popt/popthelp.o
gcc -I. -I. -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -c popt/poptparse.c -o popt/poptparse.o
gcc -I./zlib -I./popt -g -O2 -DHAVE_CONFIG_H -Wall -W -pedantic-errors -Wno-pedantic  -o rsync flist.o rsync.o generator.o receiver.o cleanup.o sender.o exclude.o util1.o util2.o main.o checksum.o match.o syscall.o log.o backup.o delete.o options.o io.o compat.o hlink.o token.o uidlist.o socket.o hashtable.o usage.o fileio.o batch.o clientname.o chmod.o acls.o xattrs.o progress.o pipe.o lib/md5-asm-x86_64.o simd-checksum-x86_64.o params.o loadparm.o clientserver.o access.o connection.o authenticate.o lib/wildmatch.o lib/compat.o lib/snprintf.o lib/mdfour.o lib/md5.o lib/permstring.o lib/pool_alloc.o lib/sysacls.o lib/sysxattrs.o  zlib/deflate.o zlib/inffast.o zlib/inflate.o zlib/inftrees.o zlib/trees.o zlib/zutil.o zlib/adler32.o zlib/compress.o zlib/crc32.o popt/findme.o  popt/popt.o  popt/poptconfig.o popt/popthelp.o popt/poptparse.o -lattr -lacl -llz4 -lzstd -lxxhash -lcrypto 
sed 's;\@bindir\@;/usr/local/bin;g' <./stunnel-rsyncd.conf.in >stunnel-rsyncd.conf
Wrote: rsync.1.html
Wrote: rsync.1
Wrote: rsync-ssl.1.html
Wrote: rsync-ssl.1
Wrote: rsyncd.conf.5.html
Wrote: rsyncd.conf.5

root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync# make install

/usr/bin/mkdir -p /usr/local/bin
/usr/bin/install -c  -m 755 rsync /usr/local/bin
/usr/bin/install -c -m 755 ./rsync-ssl /usr/local/bin
/usr/bin/mkdir -p /usr/local/share/man/man1
/usr/bin/mkdir -p /usr/local/share/man/man5
if test -f rsync.1; then /usr/bin/install -c -m 644 rsync.1 /usr/local/share/man/man1; fi
if test -f rsync-ssl.1; then /usr/bin/install -c -m 644 rsync-ssl.1 /usr/local/share/man/man1; fi
if test -f rsyncd.conf.5; then /usr/bin/install -c -m 644 rsyncd.conf.5 /usr/local/share/man/man5; fi

root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync# ls

access.c        clientname.c    csprotocol.txt           generator.o       latest-year.h   options.o           rsync.1             shconfig                  trimslash.c
access.o        clientname.o    daemon-parm.awk          getfsdev.c        lib             packaging           rsync.1.html        shconfig.in               t_stub.c
aclocal.m4      clientserver.c  daemon-parm.h            getgroups.c       loadparm.c      params.c            rsync.1.md          simd-checksum-x86_64.cpp  t_unsafe.c
acls.c          clientserver.o  daemon-parm.txt          git-version.h     loadparm.o      params.o            rsync3.txt          simd-checksum-x86_64.o    uidlist.c
acls.o          cmdormsg        default-cvsignore.h      hashtable.c       log.c           pipe.c              rsync.c             socket.c                  uidlist.o
authenticate.c  compat.c        default-dont-compress.h  hashtable.o       log.o           pipe.o              rsyncd.conf.5       socket.o                  usage.c
authenticate.o  compat.o        define-from-md.awk       help-from-md.awk  m4              popt                rsyncd.conf.5.html  stunnel-rsyncd.conf       usage.o
backup.c        config.guess    delete.c                 help-rsyncd.h     main.c          prepare-source      rsyncd.conf.5.md    stunnel-rsyncd.conf.in    util1.c
backup.o        config.h        delete.o                 help-rsync.h      main.o          prepare-source.mak  rsync.h             support                   util1.o
batch.c         config.h.in     doc                      hlink.c           Makefile        progress.c          rsync.o             syscall.c                 util2.c
batch.o         config.log      Doxyfile                 hlink.o           Makefile.in     progress.o          rsyncsh.txt         syscall.o                 util2.o
byteorder.h     config.status   errcode.h                ifuncs.h          match.c         proto.h             rsync-ssl           tech_report.tex           version.h
case_N.h        config.sub      exclude.c                INSTALL.md        match.o         proto.h-tstamp      rsync-ssl.1         testhelp                  wildtest.c
checksum.c      configure       exclude.o                install-sh        maybe-make-man  README.md           rsync-ssl.1.html    testrun.c                 wildtest.txt
checksum.o      configure.ac    fileio.c                 inums.h           md2man          receiver.c          rsync-ssl.1.md      testsuite                 xattrs.c
chmod.c         configure.sh    fileio.o                 io.c              mkgitver        receiver.o          runtests.sh         tls.c                     xattrs.o
chmod.o         connection.c    flist.c                  io.h              mkproto.awk     rounding.c          SECURITY.md         TODO                      zlib
cleanup.c       connection.o    flist.o                  io.o              NEWS.md         rounding.h          sender.c            token.c
cleanup.o       COPYING         generator.c              itypes.h          options.c       rsync               sender.o            token.o

root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync# ./rsync --version

rsync  version v3.2.3-70-g9dad3721  protocol version 31
Copyright (C) 1996-2020 by Andrew Tridgell, Wayne Davison, and others.
Web site: https://rsync.samba.org/
Capabilities:
    64-bit files, 64-bit inums, 64-bit timestamps, 64-bit long ints,
    socketpairs, symlinks, symtimes, hardlinks, hardlink-specials,
    hardlink-symlinks, IPv6, atimes, batchfiles, inplace, append, ACLs,
    xattrs, optional protect-args, iconv, prealloc, stop-at, no crtimes
Optimizations:
    SIMD, asm, openssl-crypto
Checksum list:
    xxh128 xxh3 xxh64 (xxhash) md5 md4 none
Compress list:
    zstd lz4 zlibx zlib none

rsync comes with ABSOLUTELY NO WARRANTY.  This is free software, and you
are welcome to redistribute it under certain conditions.  See the GNU
General Public Licence for details.

**Comment by Marietto2008** [comment]
root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync# make install

/usr/bin/mkdir -p /usr/local/bin
/usr/bin/install -c  -m 755 rsync /usr/local/bin
/usr/bin/install -c -m 755 ./rsync-ssl /usr/local/bin
/usr/bin/mkdir -p /usr/local/share/man/man1
/usr/bin/mkdir -p /usr/local/share/man/man5
if test -f rsync.1; then /usr/bin/install -c -m 644 rsync.1 /usr/local/share/man/man1; fi
if test -f rsync-ssl.1; then /usr/bin/install -c -m 644 rsync-ssl.1 /usr/local/share/man/man1; fi
if test -f rsyncd.conf.5; then /usr/bin/install -c -m 644 rsyncd.conf.5 /usr/local/share/man/man5; fi

# cp /usr/local/bin/rsync /usr/bin

# which rsync
/usr/local/bin/rsync

# rsync --version

root@Z390-AORUS-PRO:/home/ziomario/Scaricati/rsync2# rsync --version

rsync  version v3.2.3-70-g9dad3721  protocol version 31

**Comment by WayneD** [comment]
That is totally expected.  3.2.4 hasn't been released yet, so the version that get reported (with git available) is v3.2.3 + 70 commits.
